web3/eth.py

```diff
@@ -254,7 +254,11 @@ def estimateGas(self, transaction):
             [transaction],
         )
 
-    def filter(self, filter_params):
+    def filter(self, filter_params=None, filter_id=None):
+        if filter_id and filter_params:
+            raise ValueError("Ambiguous invocation: provide either a \
```

Comment:
IIRC, we're calling this a `TypeError` elsewhere: when the wrong combination of arguments is supplied.